sklearn/metrics/_pairwise_distances_reduction/_datasets_pair.pxd.tp

```diff
@@ -41,6 +41,7 @@ cdef class DatasetsPair{{name_suffix}}:
         ITYPE_t Y_start,
         ITYPE_t i,
         ITYPE_t j,
+        ITYPE_t n_Y,
         ITYPE_t thread_num=*
```

Comment:
What is the reason to use `*` as a default, here? What is the semantic?

Reply:
I added `*` so that `surrogate_dist` could take `0` as the default for `thread_num`. Since `thread_num` is a new argument of `surrogate_dist`, I've set a default parameter to avoid impacting other calls. WDYT?